woocommerce/woocommerce
# Add Used for variations? field to csv importer

### All Submissions:






### Changes proposed in this Pull Request:


Add "Used for variations?" field to CSV importer. This will enable editing of variable product attributes that are used for variations. Currently if these attributes are edited via the importer it will delink the variations from the variable product, as the used for variations box will no longer be checked. 
I was able to update the attributes via WP-CLI and keep the variations linked by using the 'variation' field of the attribute set to `true`. and have based the edit to the code on this. To update using WP-CLI you can use the command `wp wc product update <product_id> --attributes='[{"name":"Attribute Name","visible":false,"variation":true,"options":["value1","value2","value3"}]'

Closes #25934

### How to test the changes in this Pull Request:

1. Edit the attributes on a variable product by changing "Attribute %d value" and importing a CSV file. The attribute should match the one that is Used for variations.
2. View the variable product in the edit screen, and the attribute should have updated, but the variations will be missing. They can be restored by checking the used for variations checkbox. The values that are unchanged would be restored. To show altered values the variation would also need to be updated.
3. Import CSV file with 'Attribute %d variation' set to 1 to update the variable product to use the attribute for variations.

### Other information:

* [ ] Have you written new tests for your changes, as applicable? I'm not sure how to do this
* [ ] Have you successfully run tests with your changes locally? I don't have the development environment set up



### Changelog entry
Add "Used for variations?" field to CSV importer
> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.
